### DISCUSSION TITLE
Flash offset for custom bootloader

### DISCUSSION THREAD

**Discussion by VitorBoss** [stm32duino/Arduino_Core_STM32#1986] [General]
Recently I'm working on a project which will use [sboot_stm32](https://github.com/dmitrystu/sboot_stm32) encryption, the current problem is the linker script uses `LD_FLASH_OFFSET` but I can't set this flag from PlatformIO, I had tried put the file "platform.local.txt" with the `build.flash_offset=0x2000` definition, the `-DLD_FLASH_OFFSET=0x2000` on compilation flags and on file "build_opt.h" as the wiki says.

The problem is, don't matter what I do, the offset never get applied and the elf file always point to 0x08000000 as starting address.
What I'm missing? I dug the wiki with no luck

**Comment by fpistm** [comment]
We don't support pio. You should ask to their GitHub

**Reply by VitorBoss** [reply]
No problem, but how should this be handled by Arduino IDE?

Edit: The project is compatible with arduino IDE, I use it only as last resource due to lack of proper development tools/support

**Reply by fpistm** [reply]
Which IDE you used? Legacy 1.8.x or 2.x?

**Reply by fpistm** [reply]
You can check maple or bluepill definition in the boards.txt as example.

**Reply by fpistm** [reply]
Hi @VitorBoss 
Currently working on flash_offset, see #1993 

You can use `boards.local.txt` feature:
https://arduino.github.io/arduino-cli/0.31/platform-specification/#boardslocaltxt

For your case, it should contains something like this, depending of your board:
```
GenF1.menu.pnum.GENERIC_F103C8TX.build.flash_offset=0x2000
```
And define `VECT_TAB_OFFSET` using build_opt.h

**Reply by VitorBoss** [reply]
> Which IDE you used? Legacy 1.8.x or 2.x?

I had 1.8.13 since I last used it in 2020

**Reply by VitorBoss** [reply]
Forgive me if I didn't understood correctly, but those changes(very welcomed) don't change the linker offset right?

**Reply by fpistm** [reply]
Chande made in the PR will change also the linker offset.

**Reply by VitorBoss** [reply]
That is perfect, thank you very much for your support
